tools/rosbag/src/recorder.cpp

```diff
@@ -330,6 +336,20 @@ void Recorder::doQueue(const ros::MessageEvent<topic_tools::ShapeShifter const>&
     }
 }
 
+void Recorder::doRecordLatchedTopic(const OutgoingMessage& out)
+{
+    for (std::list<OutgoingMessage>::iterator it = latched_topics_.begin(); it != latched_topics_.end(); ++it)
```

Comment:
I could do a remove_if, but writing the predicate would end up being more code than the for loop (might be better/clearer with remove_if and a lambda, but ROS doesn't support c++11)